iup/__init__.py

```diff
@@ -309,16 +274,22 @@ def to_cumulative(self, group_cols: tuple[str,] | None, last_cumulative=None):
                 estimate=pl.col("estimate") + pl.col("last_cumulative")
             ).drop("last_cumulative")
 
-        out = CumulativeUptakeData(out)
-
         return out
 
 
 class CumulativeUptakeData(UptakeData):
     """
-    Subclass of UptakeData for cumulative uptake.
+    Subclass of ValidatedUptake for cumulative uptake.
     """
 
+    def validate(self):
+        # same validations as UptakeData
+        super().validate()
+        # and also require that uptake be a proportion
+        assert (
+            self["estimate"].is_between(0.0, 1.0).all()
+        ), "cumulative uptake `estimate` must be a proportion"
```

Comment:
Incident uptake is also a proportion, although we accept the fact that it can sometimes be reported as a negative. Should we implement a similar assertion that each "estimate" in an IncidentUptakeData is between -1.0 and 1.0?

Reply:
The way I would think about this is: I know that cumulative uptake should always be between 0 and 1, and my validation of incident uptake should be that I can convert it to cumulative without error

I'll make this a separate issue

Reply:
cf #68

Reply:
works for me!